src/Exception/ApiFilterException.php

```diff
@@ -0,0 +1,11 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Exception;
+
+class ApiFilterException extends \Exception
+{
+    public static function createFromThrowable(\Throwable $exception): self
+    {
+        return new self($exception->getMessage(), 0, $exception);
```

Comment:
By this you are effectively hiding some maybe specific exception to a generic one (and the original only including as a previous). Why is this, what is the purpose to rethrow everything as `ApiFilterException`? If the purpose is to have all exceptions from the ApiFilter being recognizable and catchable by client using it, I'd prefer using common interface and various specific exceptions for cases in which it could occur inside this library.

Reply:
Well in 99% of cases (maybe even 100%) it is `Invalid argument exception` from failed `Assertion`. 

I dont see any advantage from creating a specific exception for this case over generic `ApiFilterException` which has original exception, if you need it.

Reply:
Well, again the original question: what is the purpose of encapsulating all exceptions to a custom one?

If you want to allow the caller to catch exceptions from this library (which I believe is the reason),  then you don't need to remove the information which is in the exception type - you can have some exception for this specific case (which also expres what happened), and all of your exception could implement the common library-level interface, to allow caller catch the exceptions. 
Also by encapsulating everything to this exception, you disallows clients to act differently based on the exception type (you cannot just `catch` previous exception). This may not be the case now (as you said), but doing it this way is not future-proof.

This is also greatly explained in this article: https://blog.nikolaposa.in.rs/2016/08/17/exceptional-behavior-best-practices/ - especially "Custom Exception classes" and "Component level exceptions" parts.

Reply:
I am with @OndraM. 

```
try {
// some api filter call
} catch (ApiFilterException $e) {
}
```

won`t tell me anything. This should be at least renamed to non-generic name.